Mapify/Patches/BufferStopPatch.cs

```diff
@@ -13,49 +10,19 @@ namespace Mapify.Patches
     [HarmonyPatch(typeof(BufferStop), nameof(BufferStop.OnTriggerEnter))]
     public static class BufferStop_OnTriggerEnter_Patch
     {
-        /// <summary>
-        ///     Replaces the logic removed in the Transpiler.
-        /// </summary>
-        /// <seealso cref="Transpiler" />
-        private static bool Prefix(BufferStop __instance, Collider other)
+        private static void Prefix(BufferStop __instance, Collider other)
         {
-            Rigidbody attachedRigidbody = other.attachedRigidbody;
-            float breakSpeed = Maps.IsDefaultMap
-                ? BufferStop.SQR_BREAK_BUFFER_VELOCITY_THRESHOLD
-                : __instance.GetComponent<Editor.BufferStop>().breakSpeed * 3.6f;
-            return attachedRigidbody != null && attachedRigidbody.velocity.sqrMagnitude <= breakSpeed * breakSpeed;
-        }
+            if (Maps.IsDefaultMap) return;
 
-        /// <summary>
-        ///     Replaces the default mass with the mass.
-        /// </summary>
-        private static void Postfix(BufferStop __instance)
-        {
-            if (Maps.IsDefaultMap)
-                return;
-            Rigidbody rigidbody = __instance.gameObject.GetComponent<Rigidbody>();
-            if (rigidbody == null)
-                return;
-            rigidbody.mass = __instance.GetComponent<Editor.BufferStop>().massAfterBreak;
+            __instance.breakVelocitySqr = __instance.GetComponent<Editor.BufferStop>().breakSpeed * 3.6f;
         }
 
-        /// <summary>
-        ///     Removes the velocity check, which is replaced in our Prefix patch.
-        /// </summary>
-        /// <seealso cref="Prefix" />
-        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+        private static void Postfix(BufferStop __instance)
         {
-            List<CodeInstruction> codes = new List<CodeInstruction>(instructions);
-
-            for (int i = 0; i < codes.Count; i++)
-                if (codes[i].opcode == OpCodes.Ldc_R4 && ((float)codes[i].operand).Eq(BufferStop.SQR_BREAK_BUFFER_VELOCITY_THRESHOLD))
-                {
-                    for (int j = i - 5; j <= i + 1; j++)
-                        codes[j].opcode = OpCodes.Nop;
-                    break;
-                }
+            if (Maps.IsDefaultMap) return;
 
-            return codes;
+            //the RigidBody is created in OnTriggerEnter so we have to set rb.mass in a Postfix
+            __instance.rb.mass = __instance.GetComponent<Editor.BufferStop>().massAfterBreak;
```

Comment:
This should only run if it actually breaks, so we need to check if the rigidbody exists before setting it.

Reply:
updated